### PR TITLE
Update melos to 5.2.0

### PR DESCRIPTION
**Pull request by nielsenko**
- Use new steps feature (https://github.com/invertase/melos/pull/664)
- Drop analyze and format scripts, as these are now internal commands to melos
- Renamed format script to custom_format
- Renamed analyze:deps to lint:deps

### CODE REVIEW

**Comment by spydon** [comment]
Wow, happy to see the new feature in practice, so much cleaner!

**Comment by nielsenko** [comment]
Added no-changelog, as this is not a user visible feature, and we _sadly_ don't use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) (yet) 😄

File: melos.yaml [hunk 1]
```diff
@@ -115,7 +115,7 @@ scripts:
     packageFilters:
       published: true
 
-  format:
+  custom_format:
```
Comment (elle-j) [inline melos.yaml:118]: Was `format` conflicting with some other convention?

File: melos.yaml [hunk 2]
```diff
@@ -226,12 +217,10 @@ scripts:
...
-      melos run test &&
-      melos run coverage &&
-      melos run analyze &&
-      melos run lint
+    steps:
+      - build
+      - test
+      - coverage
+      - analyze
+      - lint
```
Comment (elle-j) [inline melos.yaml:225]: How come the `clean` and `bootstrap` commands are not explicitly needed in this case anymore?

**Review by elle-j — APPROVED** [review]
Much cleaner with `step`s 🙂

File: melos.yaml (hunk 2 above)
Comment (nielsenko) [reply melos.yaml:225]: The new steps feature is for chaining _scripts_. Currently it doesn't support _commands_ (made a comment about it [here](https://github.com/invertase/melos/issues/653#issuecomment-2008977433)). Thinking about it a bit, it was perhaps a bit much to require a `clean` anyway, so I removed that, and hence `bootstrap` is not needed either.

File: melos.yaml (hunk 1 above)
Comment (nielsenko) [reply melos.yaml:118]: Yes, as of 5.1.0 `format` is a build-in command in melos. I made an issue on it [here](https://github.com/invertase/melos/issues/661), but I guess we will at most get a warning. Anyway, the default `melos format` is a bit too simple for us (currently), hence the re-name.

File: melos.yaml (hunk 2 above)
Comment (elle-j) [reply melos.yaml:225]: Wouldn't it be necessary in some cases to clean? Or why do you think it isn't needed?

File: melos.yaml (hunk 1 above)
Comment (elle-j) [reply melos.yaml:118]: Gotcha 👍

File: melos.yaml (hunk 2 above)
Comment (nielsenko) [reply melos.yaml:225]: It should be rare, but [`melos clean`](https://melos.invertase.dev/commands/clean) basically brings us closer to a fresh start.

Comment (nielsenko) [reply melos.yaml:225]: We are not currently using the melos _ci_ script on our actual CI. It is intended to go through all (/most) of the motions the github actions does, but locally.. but to be honest it is work in progress.

In some ways it does more, in some ways it does less.

Comment (elle-j) [reply melos.yaml:225]: Oh yeah, thanks for the reminder

**Review by nirinchev — APPROVED** [review]

**Pull request merged**